src/sentry/event_manager.py

```diff
@@ -1480,7 +1480,7 @@ def get_event_type(data: Mapping[str, Any]) -> EventType:
 
 
 def materialize_metadata(
-    data: Mapping[str, Any], event_type: EventType, event_metadata: Mapping[str, Any]
```

Comment:
tbh i have no idea what the difference is

Reply:
The difference is `Mapping` doesn't have `.update`. But it's true, it could be `MutableMapping`. Given that it's an internal method, I'm honestly not sure how much it matters? But I can change it if you think it's better.